packages/odyssey-react-mui/src/Tabs.tsx

```diff
@@ -67,16 +68,29 @@ export type TabsProps = {
    * Identifier for the selected tab.
    */
   value?: string;
+  /**
+   * Callback fired when the active tab is changed.
+   */
+  onChange?: (event: SyntheticEvent, value: string) => void;
 };
 
-const Tabs = ({ ariaLabel, initialValue, tabs, value }: TabsProps) => {
+const Tabs = ({
+  ariaLabel,
+  initialValue,
+  tabs,
+  value,
+  onChange,
+}: TabsProps) => {
   const [tabState, setTabState] = useState(initialValue ?? value ?? "0");
 
-  const onChange = useCallback(
-    (_event: React.SyntheticEvent, newState: string) => {
-      setTabState(newState);
+  const handleChange = useCallback(
+    (event: SyntheticEvent, value: string) => {
+      setTabState(value);
+      if (onChange) {
+        onChange(event, value);
+      }
```

Comment:
```suggestion
  const onChange = useCallback<MuiTabListProps["onChange"]>(
    (event, value) => {
      setTabState(value);
      onChangeProp?.(event, value);
```

We should add a generic to `useCallback` so it properly types the args for us. We shouldn't need to manually type those. That method wouldn't scale.

Second, our pattern right now is naming internal functions the same as the prop (`onChange`) or something like `changeTab`.

In this case, if we name it `onChange`, we have the prop name conflicting. In that case, we leave the prop as `onChange`, but rename it in the arg to `onChangeProp`:
```ts
const Tabs: ({
  onChange: onChangeProp,
}: TabProps)
```

That's the current pattern in Odyssey.

Reply:
Had to wrap the generic useCallback type with `NonNullable` because `useCallback` doesn't like `undefined` as a function, even though it's always defined.

Reply:
Understandable. Since their type is `?`, it's possibly `undefined`. Makes sense. I wish there was a better solution, but I'm fine with this.